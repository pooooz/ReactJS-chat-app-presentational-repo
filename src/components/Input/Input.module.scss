.input_green {
  font-family: 'Press Start 2P', cursive;
  margin: 20px 0 20px 0;
  outline: none;
  border: none;
  color: limegreen;
  background: transparent;
}

.input_wrap:before {
  font-family: 'Press Start 2P', cursive;
  content: '>';
  color: limegreen
}