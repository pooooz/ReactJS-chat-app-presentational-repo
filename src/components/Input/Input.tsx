import React, { FC } from 'react';

import styles from './Input.module.scss';

interface InputProps {
  value: string;
  change: (event: React.ChangeEvent<HTMLInputElement>) => void;
  type: string;
}

export const Input: FC<InputProps> = (props) => {
  return (
    <div className={styles.input_wrap}>
      <input
        className={styles.input_green}
        type={props.type}
        value={props.value}
        onChange={props.change}
        autoFocus
      />
    </div>
  );
};
