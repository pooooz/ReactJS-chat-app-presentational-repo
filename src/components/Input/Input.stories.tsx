import React, { useState } from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Input } from './Input';

export default {
  title: 'components/Input',
  component: Input,
} as ComponentMeta<typeof Input>;

export const Default: ComponentStory<typeof Input> = () => {
  const [value, setValue] = useState('Basic value');
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setValue(event.target.value);
  };
  return <Input value={value} change={handleChange} />;
};
