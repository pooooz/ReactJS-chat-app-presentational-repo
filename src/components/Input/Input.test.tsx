import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Input } from './Input';

describe('Input', () => {
  const changeHandler = jest.fn();

  it('Render component', () => {
    render(<Input value="Render test" change={changeHandler} type="text" />);
  });

  it('Render with snapshot', () => {
    const { asFragment } = render(
      <Input value="Test" change={changeHandler} type="text" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Multiple render', () => {
    render(
      <>
        <Input value="Test" change={changeHandler} type="text" />
        <Input value="Test" change={changeHandler} type="text" />
      </>
    );
    expect(screen.getAllByRole('textbox').length).toBe(2);
  });

  it('Render with value and change handler', () => {
    render(<Input value="Test" change={changeHandler} type="text" />);
    fireEvent.input(screen.getByRole('textbox'), {
      target: { value: 'Test' },
    });
    expect(screen.getByDisplayValue(/Test/)).toBeInTheDocument();
  });
});
