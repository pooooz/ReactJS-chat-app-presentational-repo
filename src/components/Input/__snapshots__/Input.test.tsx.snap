// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="input_wrap"
  >
    <input
      class="input_green"
      type="text"
      value="Test"
    />
  </div>
</DocumentFragment>
`;
